pkgs/by-name/pe/pesign/package.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, pkg-config
+, nss
+, efivar
+, util-linux
+, popt
+, nspr
+, mandoc
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pesign";
+  version = "116";
+
+  src = fetchFromGitHub {
+    owner = "rhboot";
+    repo = "pesign";
+    rev = version;
+    hash = "sha256-cuOSD/ZHkilgguDFJviIZCG8kceRWw2JgssQuWN02Do=";
+  };
+
+  # nss-util is missing because it is already contained in nss
+  # Red Hat seems to be shipping a separate nss-util:
+  # https://centos.pkgs.org/7/centos-x86_64/nss-util-devel-3.44.0-4.el7_7.x86_64.rpm.html
+  # containing things we already have in `nss`.
+  # We can ignore all the errors pertaining to a missing
+  # nss-util.pc I suppose.
+  buildInputs = [ efivar util-linux nss popt nspr mandoc ];
+  nativeBuildInputs = [ pkg-config ];
+
+  makeFlags = [ "INSTALLROOT=$(out)" ];
+
+  postInstall = ''
+    mv $out/usr/bin $out/bin
+    mv $out/usr/share $out/share
+
+    rm -rf $out/usr
+    rm -rf $out/etc
+    rm -rf $out/run
+  '';
+
+  meta = with lib; {
+    description = "Signing tools for PE-COFF binaries. Compliant with the PE and Authenticode specifications.";
+    homepage = "https://github.com/rhboot/pesign";
+    license = licenses.gpl2Only;
+    maintainers = with maintainers; [ raitobezarius ];
```

Comment:
```suggestion
    maintainers = with maintainers; [ raitobezarius ];
    platforms = platforms.linux;
```

Reply:
Is this true though? (not that I can try building it on macos)
But that binary should be platform independent.

Reply:
This field is required anyway for the package to be cached on aarch64-linux (see [ofborg-eval](https://gist.github.com/GrahamcOfBorg/f8dd2b5faacad088d14003eb72a6cac5)). It could probably be changed to `platforms.darwin` later on.

Reply:
If someone is curious, this currently fails on the `efivar` dependency anyways, which also is Linux only.

I tried to quickly throw something together:

```
diff --git a/pkgs/tools/system/efivar/default.nix b/pkgs/tools/system/efivar/default.nix
index 8507c7c7cec0..aace09b50fac 100644
--- a/pkgs/tools/system/efivar/default.nix
+++ b/pkgs/tools/system/efivar/default.nix
@@ -1,4 +1,24 @@
-{ lib, stdenv, buildPackages, fetchFromGitHub, fetchpatch, pkg-config, popt, mandoc }:
+{ lib, stdenv, buildPackages, fetchFromGitHub, fetchpatch, pkg-config, popt, mandoc, writeTextFile }:
+
+let
+
+  # As part of building a hostPlatform=mips kernel, Linux creates and runs a
+  # tiny utility `arch/mips/boot/tools/relocs_main.c` for the buildPlatform.
+  # This utility references a glibc-specific header `byteswap.h`.  There is a
+  # compatibility header in gnulib for most BSDs, but not for Darwin, so we
+  # synthesize one here.
+  darwin-byteswap-h = writeTextFile {
+    name = "byteswap-h";
+    text = ''
+      #pragma once
+      #include <libkern/OSByteOrder.h>
+      #define bswap_16 OSSwapInt16
+      #define bswap_32 OSSwapInt32
+      #define bswap_64 OSSwapInt64
+    '';
+    destination = "/include/byteswap.h";
+  };
+in
 
 stdenv.mkDerivation rec {
   pname = "efivar";
@@ -31,7 +51,23 @@ stdenv.mkDerivation rec {
     })
   ];
 
-  nativeBuildInputs = [ pkg-config mandoc ];
+  postPatch = lib.optionalString stdenv.isDarwin ''
+    substituteInPlace src/efivar_endian.h \
+      --replace "<endian.h>" "<machine/endian.h>"
+    substituteInPlace src/include/efivar/efivar.h \
+      --replace "<endian.h>" "<machine/endian.h>"
+    substituteInPlace src/util.h \
+      --replace "<endian.h>" "<machine/endian.h>"
+    substituteInPlace src/guid.h \
+      --replace "<endian.h>" "<machine/endian.h>"
+  '';
+
+  NIX_CFLAGS_COMPILE = toString([
+    "-Wno-error"
+    "-Wno-incompatible-pointer-types"
+  ]);
+
+  nativeBuildInputs = [ pkg-config mandoc ] ++ lib.optional stdenv.isDarwin darwin-byteswap-h;
   buildInputs = [ popt ];
   depsBuildBuild = [ buildPackages.stdenv.cc ];
 
@@ -42,12 +78,12 @@ stdenv.mkDerivation rec {
     "mandir=$(man)/share/man"
     "includedir=$(dev)/include"
     "PCDIR=$(dev)/lib/pkgconfig"
-  ];
+  ] ++ lib.optional stdenv.isDarwin "CC=clang";
 
   meta = with lib; {
     description = "Tools and library to manipulate EFI variables";
     homepage = "https://github.com/rhboot/efivar";
-    platforms = platforms.linux;
+    platforms = platforms.unix;
     license = licenses.lgpl21Only;
   };
 }
```

This builds a bit further, next error are a missing [AT_EMPTY_PATH](https://patchwork.ozlabs.org/project/qemu-devel/patch/20120224195143.GA16353@vostro.hallyn.com/), as well as a missing [strchrnul](https://bugs.launchpad.net/pcb/+bug/1542858) function.

I lost patience, but in case someone wants to pick this up :-)